compiler/next/test/frontend/testParseEnums.cpp

```diff
@@ -0,0 +1,328 @@
+  assert(parseResult.topLevelExpressions[0]->isModuleDecl());
+  auto module = parseResult.topLevelExpressions[0]->toModuleDecl()->module();
+  assert(module->numStmts() == 1);
+  auto enumDecl = module->stmt(0)->toEnumDecl();
+  assert(enumDecl);
+  assert(enumDecl->numDeclOrComments() == 2);
+  auto a = enumDecl->declOrComment(0)->toEnumElementDecl();
+  auto b = enumDecl->declOrComment(1)->toEnumElementDecl();
+  checkTest4(enumDecl, a, b);
+}
+
+static void test4g(Parser* parser) {
+  auto parseResult = parser->parseString("test4g.chpl",
+                                         "enum myEnum { a = ii /* c */, b }\n");
+  assert(parseResult.errors.size() == 0);
+  assert(parseResult.topLevelExpressions.size() == 1);
+  assert(parseResult.topLevelExpressions[0]->isModuleDecl());
+  auto module = parseResult.topLevelExpressions[0]->toModuleDecl()->module();
+  assert(module->numStmts() == 1);
+  auto enumDecl = module->stmt(0)->toEnumDecl();
+  assert(enumDecl);
+  assert(enumDecl->numDeclOrComments() == 2);
+  auto a = enumDecl->declOrComment(0)->toEnumElementDecl();
+  auto b = enumDecl->declOrComment(1)->toEnumElementDecl();
+  checkTest4(enumDecl, a, b);
+}
+
+static void test4h(Parser* parser) {
+  auto parseResult = parser->parseString("test4h.chpl",
+                                         "enum myEnum { a = ii, /* c */ b }\n");
+  assert(parseResult.errors.size() == 0);
+  assert(parseResult.topLevelExpressions.size() == 1);
+  assert(parseResult.topLevelExpressions[0]->isModuleDecl());
+  auto module = parseResult.topLevelExpressions[0]->toModuleDecl()->module();
+  assert(module->numStmts() == 1);
+  auto enumDecl = module->stmt(0)->toEnumDecl();
+  assert(enumDecl);
+  assert(enumDecl->numDeclOrComments() == 3);
+  auto a = enumDecl->declOrComment(0)->toEnumElementDecl();
+  assert(enumDecl->declOrComment(1)->isComment());
+  auto b = enumDecl->declOrComment(2)->toEnumElementDecl();
+  checkTest4(enumDecl, a, b);
+}
+
+static void test4i(Parser* parser) {
+  auto parseResult = parser->parseString("test4i.chpl",
+                                         "enum myEnum { a = ii, b /* c */ }\n");
+  assert(parseResult.errors.size() == 0);
+  assert(parseResult.topLevelExpressions.size() == 1);
+  assert(parseResult.topLevelExpressions[0]->isModuleDecl());
+  auto module = parseResult.topLevelExpressions[0]->toModuleDecl()->module();
+  assert(module->numStmts() == 1);
+  auto enumDecl = module->stmt(0)->toEnumDecl();
+  assert(enumDecl);
+  assert(enumDecl->numDeclOrComments() == 3);
+  auto a = enumDecl->declOrComment(0)->toEnumElementDecl();
+  auto b = enumDecl->declOrComment(1)->toEnumElementDecl();
+  assert(enumDecl->declOrComment(2)->isComment());
+  checkTest4(enumDecl, a, b);
+}
+
+static void test4j(Parser* parser) {
+  auto parseResult = parser->parseString("test4i.chpl",
+                                         "enum myEnum { a = ii, b, /*c*/ }\n");
+  assert(parseResult.errors.size() == 0);
```

Comment:
Shouldn't this be an error?

Reply:
Nope, we allow trailing commas in our enums, following C. It works in the regular compiler too.